hips/hip-9876.md

```diff
@@ -0,0 +1,31 @@
+---
+hip: "9876"
+title: "S3 storage"
+authors: [ "Kamil Krzywicki <kamil.krzywicki@dbschenker.com>", "Björn Wenzel" ]
+created: "2025-04-08"
+type: "feature"
+status: "draft"
+helm-version: "4"
+---
+
+## Abstract
+
+Store helm release data in S3-compatible storage.
+
+## Motivation
+
+Helm release secrets can be pretty huge if release that is managed by them is big as well (long yamls/json/properties for applications). This makes our etcd storage quite big as well and this causes increased costs to handle bigger etcd nodes. 
+
+## Rationale
+
+There is an alternatve to default storage in kubernetes secrets there is database storage. But this seems to be an overkill to store simple data like helm releases with all the maintenance costs for running database with HA. 
+That's why something simple, highly available and popular like S3 (compatible) storage should be a perfect match. 
+Alternative could be pluggable storage for helm, but S3-comapatible storage seems to be very popular and could be introduced in the base helm binary.
```

Comment:
I think a better solution here is make some small changes to the OCI client to read files instead of registries in OCI layout format.

See https://oras.land/blog/lightweight-cloud-registry-oras/